### ISSUE TITLE
Diff lines starting with `---` erroneously show up as names of modified files

### ISSUE THREAD

**Issue by jfly** [so-fancy/diff-so-fancy#341]
With `changeHunkIndicators = false` in my `gitconfig` (note the "modified: Put some stuff in the" which makes it look like two files were modified):

```
$ echo "diff --git a/users.sql b/users.sql
index 471178f..1ae0a5e 100644
--- a/users.sql
+++ b/users.sql
@@ -1,2 +1,2 @@
--- Put some stuff in the users table
-INSERT INTO users ...;
+-- Put some stuff in the new_users table
+INSERT INTO new_users ...;
" | diff-so-fancy
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: users.sql
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@@ -1,2 +1,2 @@
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: Put some stuff in the 
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+-- Put some stuff in the new_users table
+INSERT INTO new_users ...;
```

With `changeHunkIndicators = true` in my `gitconfig`:

```
$ echo "diff --git a/users.sql b/users.sql
index 471178f..1ae0a5e 100644
--- a/users.sql
+++ b/users.sql
@@ -1,2 +1,2 @@
--- Put some stuff in the users table
-INSERT INTO users ...;
+-- Put some stuff in the new_users table
+INSERT INTO new_users ...;
" | diff-so-fancy
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: users.sql
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ users.sql:1 @
--- Put some stuff in the users table
-INSERT INTO users ...;
+-- Put some stuff in the new_users table
+INSERT INTO new_users ...;
```

@FatBoyXPC and I poked around a bit, and it looks like when `change_hunk_indicators` is enabled, we [sometimes set `in_hunk` to true], which prevents us from going into [the code that prints filenames bounded by horizontal rules]. However, if `change_hunk_indicators` is disabled, then we never set `in_hunk` to true, which means we can go into [the code that prints filenames bounded by horizontal rules].

We tried to capture the above behavior in a test, but it looks like there's [no way to run tests with non-default config values](https://github.com/so-fancy/diff-so-fancy/blob/57b85e395de050ce88362ed80b0024713d4529fb/diff-so-fancy#L403-L406).

[sometimes set `in_hunk` to true]: https://github.com/so-fancy/diff-so-fancy/blob/57b85e395de050ce88362ed80b0024713d4529fb/diff-so-fancy#L229
[the code that prints filenames bounded by horizontal rules]: https://github.com/so-fancy/diff-so-fancy/blob/57b85e395de050ce88362ed80b0024713d4529fb/diff-so-fancy#L184

**Comment by scottchiefbaker** [comment]
Couple of things here:

1. The example you give isn't 100% real world because there are no ANSI codes before it gets to diff-so-fancy. Can you try again using our [bug reporting process](https://github.com/so-fancy/diff-so-fancy/blob/master/reporting-bugs.md)?
2. I am unable to recreate what you're seeing with `$change_hunk_indicators` set to true with the latest version from `master`
3. If you want to run unit tests with non-default values just comment out [line 405](https://github.com/so-fancy/diff-so-fancy/blob/57b85e395de050ce88362ed80b0024713d4529fb/diff-so-fancy#L405).

![d-s-f](https://user-images.githubusercontent.com/3429760/61805126-1226dc80-adea-11e9-88aa-1a32d1af71f7.png)

If I can recreate it I'll gladly fix this issue.

**Comment by FatBoyXPC** [comment]
The ANSI codes are missing because we noticed that they were different when we generated the fixtures, and wanted to rule that out as being the reason why things were behaving differently. We also noticed it _appears_ like dsf handles diffs without color color codes, so we figured it was the safer bet at the time (this was before we realized the hunk indicators were the culprit).

Put this into `users.sql` and commit it:
```sql
-- Put some stuff in the users table
INSERT INTO users ...;
```

Update `users` to `new_users` and commit the change (at least the line with the SQL comment). Then run `git show` and you'll see the bug.

**Comment by jfly** [comment]
Here's the base64 output of running the `report-bug.sh` script, hopefully that helps! Sorry, I would have done that originally, but didn't see any mention of it when filing the issue. If you want to point future bug reporters to that script, you could consider setting up an issue template for bug reports, see https://help.github.com/en/articles/about-issue-and-pull-request-templates#issue-templates for instructions (apologies if you know about this already).

Z2l0IGRpZmYgSEVBRC4uSEVBRF4KZGlmZiAtLWdpdCBhL3VzZXJzLnNxbCBiL3VzZXJzLnNxbApp
bmRleCAxYWUwYTVlLi40NzExNzhmIDEwMDY0NAotLS0gYS91c2Vycy5zcWwKKysrIGIvdXNlcnMu
c3FsCkBAIC0xLDIgKzEsMiBAQAotLS0gUHV0IHNvbWUgc3R1ZmYgaW4gdGhlIG5ld191c2VycyB0
YWJsZQotSU5TRVJUIElOVE8gbmV3X3VzZXJzIC4uLjsKKy0tIFB1dCBzb21lIHN0dWZmIGluIHRo
ZSB1c2VycyB0YWJsZQorSU5TRVJUIElOVE8gdXNlcnMgLi4uOwoKCgpnaXQgZGlmZiBIRUFELi5I
RUFEXiAtLWNvbG9yChtbMzg7NTsxMzBtZGlmZiAtLWdpdCBhL3VzZXJzLnNxbCBiL3VzZXJzLnNx
bBtbbQobWzM4OzU7MTMwbWluZGV4IDFhZTBhNWUuLjQ3MTE3OGYgMTAwNjQ0G1ttChtbMzg7NTsx
MzBtLS0tIGEvdXNlcnMuc3FsG1ttChtbMzg7NTsxMzBtKysrIGIvdXNlcnMuc3FsG1ttChtbMTsy
OzM4OzU7MzNtQEAgLTEsMiArMSwyIEBAG1ttChtbMTszODs1OzE5Nm0tLS0gUHV0IHNvbWUgc3R1
ZmYgaW4gdGhlIG5ld191c2VycyB0YWJsZRtbbQobWzE7Mzg7NTsxOTZtLUlOU0VSVCBJTlRPIG5l
d191c2VycyAuLi47G1ttChtbMTszODs1OzM0bSsbW20bWzE7Mzg7NTszNG0tLSBQdXQgc29tZSBz
dHVmZiBpbiB0aGUgdXNlcnMgdGFibGUbW20KG1sxOzM4OzU7MzRtKxtbbRtbMTszODs1OzM0bUlO
U0VSVCBJTlRPIHVzZXJzIC4uLjsbW20KZ2l0IGNvbmZpZyBwYWdlci5kaWZmCmdpdCBjb25maWcg
cGFnZXIuc2hvdwo=

**Comment by jfly** [comment]
> I am unable to recreate what you're seeing with $change_hunk_indicators set to true with the latest version from master

Right, that's what I'm seeing as well (with `diff-so-fancy` version 1.2.6). I experience the problem with `changeHunkIndicators = false`

**Comment by scottchiefbaker** [comment]
OK I think I figured it out... can you check the latest checkin on the `next` branch.

**Comment by jfly** [comment]
Looks good!

**Comment by scottchiefbaker** [comment]
Thanks for testing...
